WordPress/gutenberg
# Rewrite URL (Permalink) panel as a popover

## What?
Rewrites the _Permalink_ panel to be a _URL_ popover in the _Summary_ (formerly _Status & visibility_) panel.

## Why?
Part of https://github.com/WordPress/gutenberg/issues/39082.

## How?
- Pretty similar to https://github.com/WordPress/gutenberg/pull/41925.
- Removed `PostLink` component from `@wordpress/edit-post`.
- Added `PostURL` component to `@wordpress/edit-post` and `@wordpress/editor`.
  - This is the same file structure we use for other rows in the _Summary_ panel e.g. `PostSchedule` and `PostVisibility`.
  - All of the logic (and most of the code, tbh) is the same as it was in `PostLink`.

## Testing Instructions
1. Create or edit a post or a page.
2. Open the sidebar and change the post or page's URL. Play around with editing URLs. 
3. Check that you can't edit the URL if the post type doesn't support it or if user doesn't permission.

## Screenshots or screencast 
https://user-images.githubusercontent.com/612155/176372023-3b302be3-262b-4d68-b134-af63ba0fe320.mp4
